### PR TITLE
fix: #1188 [Snapshot] Render canonicalUrl in elementdefinition.constr…

### PR DESCRIPTION
**Pull request by mmsmits**
#1188 [Snapshot] Render canonicalUrl in elementdefinition.constraint.source instead of FhirType name

### CODE REVIEW

File: src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGenerator.cs [hunk 1]
```diff
@@ -771,6 +775,18 @@ void createNewElement(ElementDefinitionNavigator snap, ElementDefinitionNavigato
             mergeElement(snap, diff);
         }
 
+        private static void AddConstraintSource(ElementDefinition targetElement, string url)
+        {
+
+            if (targetElement != null && targetElement?.Constraint.Any(c => string.IsNullOrEmpty(c.Source)) == true)
+            {
+                foreach (var constraint in targetElement.Constraint.Where(c => string.IsNullOrEmpty(c.Source)))
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGenerator.cs]:
```suggestion
                foreach (var constraint in targetElement?.Constraint.Where(c => string.IsNullOrEmpty(c.Source)) ?? Enumeration.Empty())
```

Comment (marcovisserFurore) [reply src/Hl7.Fhir.Specification/Specification/Snapshot/SnapshotGenerator.cs]: The `if` statement is not needed here.

**Comment by marcovisserFurore** [comment]
@mmsmits : is this only meant for r4?

**Comment by mmsmits** [comment]
> 
> 
> @mmsmits : is this only meant for r4?

Yes! In Stu3 only the type name was rendered

**Pull request merged**